### PR TITLE
[ENH] Simplify API system

### PR DESCRIPTION
**Pull request by 36000**
- [x] remove _file from pimms efferents
- [x] replace attributes and attached methods with myafq.export
- [x] remove need to put in arg names manually when decorating with as_file or other decorators
- [x] Other minor changes to simplify code
- [x] update docs

### CODE REVIEW

**Comment by 36000** [comment]
@arokem  would it be OK to move away from 3.7, this PR requires some syntax from 3.8 . We could also add in 3.9 or 3.10 to tests?

**Comment by arokem** [comment]
Yeah - I'd be fine requiring >=3.8

**Comment by 36000** [comment]
Note to self: rebase on #827 and use that new function before merging.

**Comment by arokem** [comment]
Looks like there's some small conflicts related to the changes in #828, so might need to rebase and resolve before we can merge this.

**Comment by arokem** [comment]
This now needs another rebase after merging #829

**Comment by 36000** [comment]
@arokem  this is ready for review / merge

**Review by arokem — COMMENTED** [review]
Looks good overall. I had one small question/comment.

File: AFQ/api/utils.py [hunk 1]
```diff
@@ -73,30 +74,53 @@
...
+        if seg_algo == "AFQ":
+            api_afq_object.export("rois")
+    api_afq_object.export("sl_counts")
+    api_afq_object.export("profiles")
+
+    if viz:
+        try:
+            api_afq_object.export("tract_profile_plots")
+        except ImportError as e:
+            plotly_err_message = viz_import_msg_error("plot")
```
[the remark this answers is not in the document]
Comment (36000) [reply AFQ/api/utils.py:119]: No, "plot" is different than "plotly". "plotly" is installed by default, these are the "plot" installs:
```
plot =
    pingouin>=0.3
    seaborn>=0.11.0
    ipython>=7.13.0,<=7.20.0
```
We use these to, for example, make the ICC plots after running tractometry. It's possible I should work on re-organizing the plot code and moving it to AFQ-insight

[the remark this answers is not in the document]
Comment (arokem) [reply AFQ/api/utils.py:119]: Gotcha. That makes sense. Then maybe this line (I believe this also appears in other places) should be: 
```suggestion
            plot_err_message = viz_import_msg_error("plot")
```

**Pull request merged**